packages/examples/cvat/exchange-oracle/src/crons/process_job_launcher_webhooks.py

```diff
@@ -91,20 +70,28 @@ def handle_job_launcher_event(webhook: Webhook, *, db_session: Session, logger:
                 cvat.create_task(webhook.escrow_address, webhook.chain_id)
 
             except Exception as ex:
-                try:
-                    cvat.remove_task(webhook.escrow_address)
-                except Exception as ex_remove:
-                    logger.exception(ex_remove)
+                projects = cvat_db_service.get_projects_by_escrow_address(
+                    db_session, webhook.escrow_address
+                )
+
+                EscrowCleaner(
+                    webhook.escrow_address, Networks(webhook.chain_id), projects
+                ).cleanup()
+
+                cvat_db_service.delete_projects(db_session, [project.id for project in projects])
 
+                # We should not notify before the webhook handling attempts have expired
                 if webhook.attempts + 1 >= Config.webhook_max_retries:
-                    # We should not notify before the webhook handling attempts have expired
-                    oracle_db_service.outbox.create_webhook(
-                        session=db_session,
-                        escrow_address=webhook.escrow_address,
-                        chain_id=webhook.chain_id,
-                        type=OracleWebhookTypes.job_launcher,
-                        event=ExchangeOracleEvent_TaskCreationFailed(reason=str(ex)),
-                    )
+                    # new session is required since exception might roll back the current session
+                    # creation of this webhook should not depend on this
+                    with SessionLocal.begin() as session:
```

Comment:
I've added `on_fail` parameter to `handle_webhook` to pass `src.crons.webhooks.job_launcher.handle_failure` which will be called in case of exception. This way `handle_webhook` also is able to abstract session handling away from callee.